test/with_dummyserver/test_connectionpool.py

```diff
@@ -452,7 +452,6 @@ def test_socket_timeout_updated_on_reuse_parameter(
             )
 
     def test_tunnel(self, http_version: str) -> None:
-        # note the actual httplib.py has no tests for this functionality
```

Comment:
This has been tested for 10 years now! 5 years after urllib3's first commit. https://github.com/python/cpython/blob/e001027188001f4bdf6ea16f70726ca0fabe85c4/Lib/test/test_httplib.py#L2225